JavaScript/4-binarySearchTree.js

```diff
@@ -0,0 +1,188 @@
+'use strict';
+
+function Node(parent, data) {
+  this.data = data;
+  this.leftChild = null;
+  this.rightChild = null;
+  this.parent = parent;
+}
+
+Node.prototype.output = function() {
+  if (this.leftChild !== null) {
+    this.leftChild.output();
+  }
+  console.dir(this.data);
```

Comment:
I think it should be `console.log()` instead of `console.dir()`. This works too, yeah, but `console.log()` would be more idiomatic.